### PR TITLE
[native] Add config to disable async data cache

### PR DESCRIPTION
**Pull request by pranjalssh**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by tanjialiang** [comment]
Thanks Pranjal. We need to do some additional validations/enforcement for disabling AsyncDataCache. when async cache is disabled, ssd has to be disabled, as async cache is a pre-requisite for ssd cache.

A verification run might be nice: disable async cache, run some queries, stop running queries. See if memory footprint drops to original state.

File: presto-native-execution/presto_cpp/main/PrestoServer.cpp [hunk 1]
```diff
@@ -456,19 +456,22 @@ void PrestoServer::initializeVeloxMemory() {
...
+    allocator_ = std::make_shared<memory::MmapAllocator>(options);
   } else {
-    allocator = memory::MemoryAllocator::createDefaultInstance();
+    allocator_ = memory::MemoryAllocator::createDefaultInstance();
   }
-  cache_ = std::make_shared<cache::AsyncDataCache>(
-      allocator, memoryBytes, std::move(ssd));
-  memory::MemoryAllocator::setDefaultInstance(cache_.get());
+  if (systemConfig->asyncDataCacheEnabled()) {
+    auto cache = std::make_shared<cache::AsyncDataCache>(
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/PrestoServer.cpp:469]:
```
cache_ = std::make_shared<cache::AsyncDataCache>(
  memory::MemoryAllocator::setDefaultInstance(cache_.get());	        allocator_, memoryBytes, std::move(ssd));
allocator_ = cache_;
```

File: presto-native-execution/presto_cpp/main/PrestoServer.h [hunk 2]
```diff
@@ -132,9 +132,11 @@ class PrestoServer {
   // Executor for async IO for connectors.
   std::unique_ptr<folly::IOThreadPoolExecutor> connectorIoExecutor_;
 
-  // Instance of AsyncDataCache used for all large allocations.
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/PrestoServer.h:135]:
```
// If not null,  the instance of AsyncDataCache used for in-memory file cache.
```

File: presto-native-execution/presto_cpp/main/PrestoServer.h [hunk 3]
```diff
@@ -132,9 +132,11 @@ class PrestoServer {
...
   std::shared_ptr<velox::cache::AsyncDataCache> cache_;
 
+  // Instance of MemoryAllocator used for all large allocations.
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/PrestoServer.h:137]:
```
// Instance of MemoryAllocator used for all query memory allocations.
//
// NOTE: AsyncDataCache implements MemoryAllocator interface and wraps on top of a real memory allocator for the actual memory allocation. So if 'cache_' has been set, 'allocator_' is also set to 'cache_'. It provides memory allocations for both file cache and query memory.
```

**Comment by pranjalssh** [comment]
I ran queries and check that presto_cpp.presto_cpp.memory_cache_total_large_bytes.avg.60 is always 0.

File: presto-native-execution/presto_cpp/main/PrestoServer.cpp [hunk 4]
```diff
@@ -456,19 +456,26 @@ void PrestoServer::initializeVeloxMemory() {
...
-    allocator = std::make_shared<memory::MmapAllocator>(options);
+    allocator_ = std::make_shared<memory::MmapAllocator>(options);
   } else {
-    allocator = memory::MemoryAllocator::createDefaultInstance();
+    allocator_ = memory::MemoryAllocator::createDefaultInstance();
   }
-  cache_ = std::make_shared<cache::AsyncDataCache>(
-      allocator, memoryBytes, std::move(ssd));
-  memory::MemoryAllocator::setDefaultInstance(cache_.get());
+  if (systemConfig->asyncDataCacheEnabled()) {
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/PrestoServer.cpp:468]: Shall we skip initialize ssd cache if async data cache is not configured? We don't support to access SSD without async data cache for now. Thanks!

File: presto-native-execution/presto_cpp/main/PrestoServer.cpp [hunk 5]
```diff
@@ -446,42 +446,51 @@ void PrestoServer::initializeVeloxMemory() {
...
+    std::unique_ptr<cache::SsdCache> ssd;
+    const auto asyncCacheSsdGb = systemConfig->asyncCacheSsdGb();
+    if (asyncCacheSsdGb) {
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/PrestoServer.cpp:463]:
```
if (asyncCacheSsdGb > 0) {
```

File: presto-native-execution/presto_cpp/main/PrestoServer.cpp [hunk 6]
```diff
@@ -446,42 +446,51 @@ void PrestoServer::initializeVeloxMemory() {
...
+          kNumSsdShards,
+          cacheExecutor_.get(),
+          asyncCacheSsdCheckpointGb << 30,
+          asyncCacheSsdDisableFileCow);
+    }
+    cache_ = std::make_shared<cache::AsyncDataCache>(
+        allocator_, memoryBytes, std::move(ssd));
+    allocator_ = cache_;
+  } else {
+    VELOX_CHECK(
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/PrestoServer.cpp:488]:
```
VELOX_CHECK_EQ(
```

**Review by xiaoxmeng — APPROVED** [review]
@pranjalssh LGTM. Please make sure if run through no problem on batch cluster before land.

**Pull request merged**